middle_end/flambda/types/env/typing_env.rec.ml

```diff
@@ -195,7 +214,15 @@ end = struct
         symbol_projections
         Variable.Map.empty
     in
-    { names_to_types; aliases; symbol_projections; }
+    let depth_variables =
+      Depth_variable.Map.fold (fun dv rec_info_with_binding_time_and_mode acc ->
```

Comment:
This says "_and_mode" but in fact we're not storing the modes.  However I think we're going to need to do that; when a depth variable goes out of scope, it will need to go from `Normal` to `In_types`.

Reply:
Okay, I've added a `name_mode` parameter that gets stored along with the `rec_info`.